### PR TITLE
User stopped typing behavior extend with minimum text length property

### LINKED ISSUE #387: [Enhancement] Extend UserHasStoppedTypingBehavior with a minimum text length property
## Summary

The UserHasStoppedTypingBehavior executes a command when the user has stopped typing (or at least the configured threshold time has passed). I would like this behavior to also support a minimum text length before this command gets executed.

## API Changes

The UserHasStoppedTypingBehavior gets an extra property: MinimumTextLength of type int.

## Intended Use Case
In the case that you don't want to start searching when the user has only typed less then x characters. F.e. the minimum text length is set to 5 and the user has stopped typing after 3 characters, the command shouldn't get executed. Also the Keyboard shouldn't get dismissed in this case, so the user can easily continue his typing.

## Who Will Do The Work?

- [x] I am willing to take this on myself
- [ ] Just putting this out there for someone to pick up

### PR DESCRIPTION
**Pull request by jBijsterboschNL**
### Description of Change ###

Implemented MinimumLengthThreshold on UserStoppedTypingBehavior.

### Bugs Fixed ###

- Fixes #387

### API Changes ###

Added: 
 
- `int MinimumLengthThreshold { get; set; }` on UserStoppedTypingBehavior

### Behavioral Changes ###

There is now also an option to set the minimum number of characters, along with the time threshold option, before the Command is being executed.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/xamarin/XamarinCommunityToolkit?pullRequest=389) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/xamarin/XamarinCommunityToolkit?pullRequest=389) before we can accept your contribution.<br/><br/>:x: jBijsterboschNL [sign now](https://cla.dotnetfoundation.org/xamarin/XamarinCommunityToolkit?pullRequest=389)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/xamarin/XamarinCommunityToolkit?pullRequest=389) it.</sub>

File: XamarinCommunityToolkit.UnitTests/Behaviors/UserStoppedTypingBehavior_Tests.cs [hunk 1]
```diff
@@ -68,13 +69,84 @@ public async Task KeyboardDimissesAfterTimeThresholdExpires()
...
+		/// <returns></returns>
+		//[Fact]
+		//public async Task ShouldNotDismissKeyboardWhenMinimumLengthThreholdHasNotBeenReached()
+		//{
+		//	// arrange
+		//	var entry = CreateEntryWithBehavior(lengthThreshold: 3,
+		//										shouldDismissKeyboardAutomatically: true);
+
+		//	// act
+		//	entry.Focus();
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit.UnitTests/Behaviors/UserStoppedTypingBehavior_Tests.cs:125]: Can we try to work around it anyhow?

```csharp
entry.SetValue(Entry.IsFocusedProperty, true);
```

or

Invoking it in MainThread

File: XamarinCommunityToolkitSample/Resx/AppResources.Designer.cs [hunk 2]
```diff
@@ -10,35 +10,48 @@
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkitSample/Resx/AppResources.Designer.cs:10]: @jfversluis can we add this file to "git ignore"?

That's always regenerated with hundreds of lines



Can we discard this file?

Comment (jfversluis) [reply XamarinCommunityToolkitSample/Resx/AppResources.Designer.cs:10]: Yeah I've looked into that. Not sure if that will work OK every time with regenerating on the developer side, but I guess it's worth a try

File: XamarinCommunityToolkit.UnitTests/Behaviors/UserStoppedTypingBehavior_Tests.cs (hunk 1 above)
Comment (jBijsterboschNL) [reply XamarinCommunityToolkit.UnitTests/Behaviors/UserStoppedTypingBehavior_Tests.cs:125]: Hi @AndreiMisiukevich,

I'd already tried to invoke it on the MainThread, but then I'm getting the following error. And I don't know 100% sure, but you can only call the Xamarin.Forms.Init() only when running on a specific platform, which is not possible in a unit test, right (or at least not that easily)?

`System.InvalidOperationException : You MUST call Xamarin.Forms.Init(); prior to using it.`

When calling the SetValue on the entry field, I'm getting this error. Which makes sense as well.

`The BindableProperty "IsFocused" is readonly.`

Comment (AndreiMisiukevich) [reply XamarinCommunityToolkit.UnitTests/Behaviors/UserStoppedTypingBehavior_Tests.cs:125]: Let's remove this commented code then

**Review by AndreiMisiukevich — APPROVED** [review]

**Review by jfversluis — APPROVED** [review]

**Comment by AndreiMisiukevich** [comment]
@jfversluis @jBijsterboschNL I am going to try to work around the "IsFocused" issue in the follow-up PR (If I was lucky enough with this problem)

For now, it's 💯 approval from my side

**Comment by jBijsterboschNL** [comment]
@AndreiMisiukevich, I would love to hear from you once you're able to work around this 'IsFocused' issue from a unit test standpoint :)

**Pull request merged**